spec/integ/provisioning.spec.js

```diff
@@ -0,0 +1,368 @@
+"use strict";
+var Promise = require("bluebird");
+var test = require("../util/test");
+var env = test.mkEnv();
+var config = env.config;
+
+describe("Provisioning API", function() {
+    var mxUser = {
+        id: "@flibble:wibble",
+        nick: "M-flibble"
+    };
+
+    var ircUser = {
+        nick: "bob",
+        localpart: config._server + "_bob",
+        id: "@" + config._server + "_bob:" + config.homeserver.domain
+    };
+
+    // Create a coroutine to test certain API parameters.
+    //  parameters {object} - the API parameters
+    //  shouldSucceed {boolean} - true if the request should succeed
+    //  link {boolean} - true if this is a link request (false if unlink)
+    let mockLink = function (parameters, shouldSucceed, link) {
+
+        return test.coroutine(function*() {
+            let json = jasmine.createSpy("json(obj)")
+            .andCallFake(function(obj) {
+                console.log('JSON ' + JSON.stringify(obj))
+            });
+            let status = jasmine.createSpy("status(num)")
+            .andCallFake(function(number) {
+                console.log(`HTTP STATUS ${number}`)
+            });
+
+            // Defaults
+            if (!parameters.matrix_room_id) {
+                parameters.matrix_room_id = "!foo:bar";
+            }
+            if (!parameters.remote_room_server) {
+                parameters.remote_room_server = "irc.example";
+            }
+            if (!parameters.remote_room_channel) {
+                parameters.remote_room_channel = "#provisionedchannel";
+            }
+
+            // When the _link promise resolves
+            let resolve = shouldSucceed ?
+                // success is indicated with empty object
+                () => { expect(json.calls[0].args[0]).toEqual({}); }:
+                // failure with 500 and JSON error message
+                () => {
+                    expect(json).toHaveBeenCalled();
+                    expect(status).toHaveBeenCalled();
```

Comment:
Look into [toHaveBeenCalledWith](http://jasmine.github.io/2.0/introduction.html#section-Spies) to reduce the number of separate assertions here.